source/retryable-writes/retryable-writes.rst

```diff
@@ -120,13 +111,22 @@ Behavioral Changes for Write Commands
 -------------------------------------
 
 Any helper method that takes a write concern parameter (see the `CRUD`_ and
-`Read and Write Concern`_ specifications) MUST automatically add a transaction
-ID when executed within a DriverSession where retryable writes have been
-enabled. The client MUST NOT check whether the specific write command supports
-retryability. If the client provides a helper method for any of the "other
-commands that write" specified in the Read and Write Concern specification, the
-method MUST automatically add a transaction ID when executed within a
-DriverSession where retryable writes have been enabled.
+`Read and Write Concern`_ specifications) MUST also accept an optional
+DriverSession parameter. If a DriverSession parameter is specified by the
+application, drivers MUST use it to generate the transaction ID for a retryable
+write operation. Otherwise, drivers MUST internally construct a new
+DriverSession for the sole purpose of generating a transaction ID. Any
+internally constructed DriverSession SHOULD be destroyed as soon as the
+operation is complete in its interactions with the server so that the
+DriverSession may return its server session to the pool.
```

Comment:
"complete in its interactions with the server" seemed like a portable way to address all drivers, since the helper method returning might not apply in the async case. If a write operation is complete in its interactions with the server, that means it either succeeded (after possible retry attempts) or we've hit an error that we're throwing back to the user.